File: web-pm-vertx-java/src/main/java/cn/yjl/vertx/handler/QueryAccountHandler.java
package cn.yjl.vertx.handler;

import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.RoutingContext;
import io.vertx.mysqlclient.MySQLPool;
import io.vertx.sqlclient.Tuple;

/**
 * @author jiaolongyin
 * @description: TODO
 * @date 2022/12/28 11:48
 */
public class QueryAccountHandler extends AbstractHandler {

    public QueryAccountHandler(MySQLPool mysqlClient) {
        super(mysqlClient);
    }

    @Override
    public void handle(RoutingContext context) {
        String id = context.queryParams().get("id");
        mysqlClient.preparedQuery("select id, created_at, updated_at, deleted_at, act_name, act_pwd, act_nick_name, act_introduction, act_status, act_register_date from accounts where id = ?")
            .execute(Tuple.of(Long.valueOf(id)))
            .map(rows -> {
                if (rows.size() > 0) {
                    return rows.iterator().next().toJson();
                } else {
                    return new JsonObject();
                }
            })
            .onSuccess(data -> context.end(data.toBuffer()));
    }
}

File: web-pm-vertx/src/main/kotlin/cn/yjl/vertx/handler/QueryAccountHandler.kt
package cn.yjl.vertx.handler

import cn.yjl.vertx.entity.AccountsEntity
import cn.yjl.vertx.util.toEntity
import io.vertx.ext.web.RoutingContext
import io.vertx.kotlin.coroutines.await
import io.vertx.sqlclient.SqlClient
import io.vertx.sqlclient.Tuple

class QueryAccountHandler(private val mysqlClient: SqlClient): AbstractHandler {

    override suspend fun handle(context: RoutingContext) {
        val id = context.queryParam("id").getOrElse(0) { "0" }
        val result = mysqlClient.preparedQuery("select * from accounts where id = ?")
            .execute(Tuple.of(id.toLong())).await()
        val rows = result.toList()
        if (rows.isNotEmpty()) {
            context.end(rows[0].toEntity(AccountsEntity()).toBuffer()).await()
        } else {
            context.end(AccountsEntity().toBuffer()).await()
        }
    }

}
